### PR TITLE
Validate form for organization contact

### LINKED ISSUE #48: ☠️ Validate registration form
**Type**: client

**FOF**: 2

## Description
- We want to check for valid form inputs before allowing the nonprofit admin to continue to the next screen

## Screenshot
image

## Todo

- [x] Check to make sure that required fields are populated
- [x] Check to make sure that structured inputs (email, phone number, etc.) are valid
- [x] Check to make sure that email addresses are unique to the nonprofit database
- [x] Check to make sure that passwords match and meet secure "requirement"

## Links

- Talk to Aditya about form validation ideas and how to check for unique emails
- [Figma](https://www.figma.com/file/qELi7Tz3K4XxXQmCVrrpRE/Clear-Path-x-H4I?node-id=1%3A2&t=wnmmIblE2fpTrpNM-0)
- [Styled Components Tips](https://javascript.plainenglish.io/react-7-tricks-to-work-with-styled-components-20eb155934b1)

## Learnings
- Formik Validation!

### PR DESCRIPTION
**Pull request by su-esther**
## Status:

:rocket: Ready

## Summary
- validates signup form for all required fields
- passwords must match, email must be available, phone number and email must be formatted correctly
Closes #48 

## Changes

- added validation in organization-contact-info
- created Yup schema for form validation
- validated email with new hook useGetEmailValidation()

## Screenshots
Screen Shot 2023-03-30 at 10 59 06 AM
Screen Shot 2023-03-30 at 10 59 52 AM

### CODE REVIEW

**Comment by su-esther** [comment]
The form validation is currently using Yup. It could also be done through setting errors into the FormikInput, similar to how email is being validated. That could be done using a lot of useState variables for each field and setting them onSubmit--but the code might be less readable?

File: src/components/organization-form-steps/organization-category/index.tsx [hunk 1]
```diff
@@ -1,64 +1,96 @@
...
   }, [setFormId]);
 
-  const handleClickOne = () => {
-    setVariantOne("active");
-    setVariantTwo("default");
-  };
-  const handleClickTwo = () => {
-    setVariantTwo("active");
-    setVariantOne("default");
+  const showVariant = (container: string) => {
```
Comment (ayanmallik) [inline src/components/organization-form-steps/organization-category/index.tsx:27]: Check out how I refactored the variant logic to only need one state variable for `category`!

File: src/components/organization-form-steps/organization-contact-info/config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as Yup from "yup";
+
+export const contactValidationSchema = Yup.object().shape({
```
Comment (ayanmallik) [inline src/components/organization-form-steps/organization-contact-info/config.ts:3]: Awesome work leveraging Formik's validation partner in Yup 😎

File: src/store/organization/api.ts [hunk 3]
```diff
@@ -53,6 +56,13 @@ export const orgApi = createApi({
         body: org,
       }),
     }),
+    getEmailValidation: builder.mutation<getEmailResponse, string>({
```
Comment (ayanmallik) [inline src/store/organization/api.ts:59]: Super cool making this mutation function that sends the email validation API request 😎

**Review by ayanmallik — APPROVED** [review]
Fantastic work Esther leveraging existing code as reference to support your email validation! I made some minor refactors regarding state management and error displaying which I can show you during dev night 💻

Let's also note that we still need to implement validation for the remaining registration screens like organization details ✍️

**Pull request merged**